stella-boleto/src/main/java/br/com/caelum/stella/boleto/bancos/BancoDoBrasil.java

```diff
@@ -18,51 +49,44 @@
 
 	private static final String NUMERO_BB = "001";
 	private static final String DIGITO_NUMERO_BB = "9";
-	private static final String CARTEIRA_16 = "16";
-	private static final String CARTEIRA_17 = "17";
-	private static final String CARTEIRA_18 = "18";
-	private static final String ZEROS_CONVENIOS_NOVOS = "000000";
-	private static final String TIPO_MODALIDADE_COBRANCA_CARTEIRA_SEM_REGISTRO = "21";
-
 
 	@Override
 	public String geraCodigoDeBarrasPara(Boleto boleto) {
-		StringBuilder campoLivre = new StringBuilder();
 		Beneficiario beneficiario = boleto.getBeneficiario();
-		
-		if (convenioAntigo(beneficiario.getNumeroConvenio())) {
-			if (beneficiario.getCarteira().equals(CARTEIRA_16) || beneficiario.getCarteira().equals(CARTEIRA_18)) {
-				campoLivre.append(getNumeroConvenioFormatado(beneficiario));
-				campoLivre.append(getNossoNumeroFormatado(beneficiario));
-				campoLivre.append(TIPO_MODALIDADE_COBRANCA_CARTEIRA_SEM_REGISTRO);
-			} else {
-				campoLivre.append(getNossoNumeroFormatado(beneficiario));
-				campoLivre.append(beneficiario.getAgenciaFormatada());
-				campoLivre.append(beneficiario.getCodigoBeneficiario());
-				campoLivre.append(boleto.getBanco().getCarteiraFormatado(beneficiario));
-			}
-		} else if (beneficiario.getCarteira().equals(CARTEIRA_17) || beneficiario.getCarteira().equals(CARTEIRA_18)) {
-			campoLivre.append(ZEROS_CONVENIOS_NOVOS);
-			campoLivre.append(getNumeroConvenioFormatado(beneficiario));
-			campoLivre.append(getNossoNumeroParaCarteiras17e18(beneficiario));
-			campoLivre.append(boleto.getBanco().getCarteiraFormatado(beneficiario));
-		} else {
-			throw new CriacaoBoletoException(
-					"Erro na geração do código de barras. Nenhuma regra se aplica. " +
-					"Verifique carteira e demais dados.");
-		}
-		return new CodigoDeBarrasBuilder(boleto).comCampoLivre(campoLivre);
-	}
-
-	private String getNossoNumeroParaCarteiras17e18(Beneficiario beneficiario) {
-		int indice = beneficiario.getCarteira().equals(CARTEIRA_17) ? 1 : 7;
-		return  getNossoNumeroFormatado(beneficiario).substring(indice);
-	}
+                String numeroConvenio = beneficiario.getNumeroConvenio();
+
+                //Exigencia do BB: numero do convenio
+                if (numeroConvenio == null 
+			|| numeroConvenio.isEmpty() ) {
+                        throw new IllegalArgumentException("O número do convênio não pode ser nulo!");
+                }                
+                //Assumimos que o desenvolvedor sabe compor o NOSSO NUMERO,
+                //Deveríamos validá-lo com base na modalidade? Hum, NÃO!!
+                int numeroPosicoesConvenio = numeroConvenio.length();
```

Comment:
no lugar de comentários no código, prefira sempre o javadoc e só nos pontos que realmente for necessário 
seres humanos não gostam de ler muito (;

sugestão:
- oq for essencial de regra confusa, vira javadoc
- oq for importante ficar claro, no lugar de comentário, criar teste de unidade.

faz sentido?

Reply:
Faz